### PR TITLE
Add PlexPy (plexpy-custom)

### PR DESCRIPTION
**Pull request by rjblake**
Note that this is still subject to DSM6.0 adduser issues. Suggest is to create plexpy-custom user through the DSM UI before installing. Packages has been tested only on DSM 6.0-7321 Update 1. Problem should not be an issue with <DSM6.0. Also note that the 'Plex Media Server logs' will not be visible in PlexPy due to plexpy-custom not having access

### CODE REVIEW

File: spk/plexpy-custom/Makefile [hunk 1]
```diff
@@ -0,0 +1,46 @@
+SPK_NAME = plexpy-custom
+SPK_VERS = $(shell date +%Y%m%d)
+SPK_REV = 4
```
Comment (cytec) [inline spk/plexpy-custom/Makefile]: `SPK_REV` starts with 1

File: spk/plexpy-custom/Makefile [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+# Pure Python package, make sure ARCH is not defined
+override ARCH=
+
+include ../../mk/spksrc.spk.mk
+
+.PHONY: plexpy-custom_extra_install
+plexpy-custom_extra_install:
+	install -m 755 -d $(STAGING_DIR)/var
+	# install -m 600 src/config.ini $(STAGING_DIR)/var/config.ini
```
Comment (cytec) [inline spk/plexpy-custom/Makefile]: if not needed remove this line

File: spk/plexpy-custom/Makefile [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	install -m 755 -d $(STAGING_DIR)/app
+	# install -m 644 src/app/config $(STAGING_DIR)/app/config
```
Comment (cytec) [inline spk/plexpy-custom/Makefile]: same here

File: spk/plexpy-custom/src/installer.sh [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    # Link
+    ln -s ${SYNOPKG_PKGDEST} ${INSTALL_DIR}
+
+    # Create a Python virtualenv
+    ${VIRTUALENV} --system-site-packages ${INSTALL_DIR}/env > /dev/null
+
+    if [ "${SYNOPKG_PKG_STATUS}" == "INSTALL" ]; then
+        # Clone the repository
+        ${GIT} clone --depth 10 --recursive -q -b ${wizard_fork_branch:=master} ${wizard_fork_url:=git://github.com/drzoidberg33/plexpy.git} ${INSTALL_DIR}/var/plexpy > /dev/null 2>&1
+        # cp ${INSTALL_DIR}/var/SickBeard/autoProcessTV/autoProcessTV.cfg.sample ${INSTALL_DIR}/var/SickBeard/autoProcessTV/autoProcessTV.cfg
```
Comment (cytec) [inline spk/plexpy-custom/src/installer.sh]: remove this

**Comment by cytec** [comment]
not sure if `sc-media` is needed here... does PlexPy require any FS access?
Besides that and the few comments above looks good to me \cc @Dr-Bean

File: spk/plexpy-custom/Makefile [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+DEPENDS  =
+SPK_DEPENDS = "python>2.7.3-3:git"
+
+MAINTAINER = RJBlake
+DESCRIPTION = PlexPy Custom allows you to run the fork of your choice of PlexPy. You will be asked for the fork\'s git URL during install.
+ADMIN_PORT = 8183
+RELOAD_UI = yes
+DISPLAY_NAME = PlexPy Custom
+CHANGELOG = "Maintenance release"
```
Comment (Dr-Bean) [inline spk/plexpy-custom/Makefile]: `Initial release` works better here.

File: spk/plexpy-custom/src/config.ini [hunk 6]
```diff
@@ -0,0 +1,3 @@
+[General]
+launch_browser = 0
+web_port = 8183
```
Comment (Dr-Bean) [inline spk/plexpy-custom/src/config.ini]: Please choose a different port. See https://github.com/SynoCommunity/spksrc/wiki/SynoCommunity-Used-Ports#how-to-choose-a-port, the `xyz` sequence you chose doesn't follow our conventions.

File: spk/plexpy-custom/src/dsm-control.sh [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+GIT="${GIT_DIR}/bin/git"
+PLEXPY="${INSTALL_DIR}/var/plexpy/PlexPy.py"
+CFG_FILE="${INSTALL_DIR}/var/config.ini"
+PID_FILE="${INSTALL_DIR}/var/plexpy-custom.pid"
+LOG_FILE="${INSTALL_DIR}/var/logs/plexpy.log"
+
+
+start_daemon ()
+{
+    su - ${USER} -s /bin/sh -c "HOME=${INSTALL_DIR}/var PATH=${PATH} ${PYTHON} ${PLEXPY} --daemon --pidfile ${PID_FILE} --config ${CFG_FILE} --datadir ${INSTALL_DIR}/var/"
```
Comment (Dr-Bean) [inline spk/plexpy-custom/src/dsm-control.sh]: Please remove the `-` after `su`.

**Comment by Dr-Bean** [comment]
Some more comments. Same question as cytec: I don't know how PlexPy builds its library, but if the filesystem isn't read, then you probably just want to use the `nobody` group, and not add the the user to `sc-media`. 
User creation for DSM6 can be added once we've finalized what that should look like, so no need to provide for that.

**Comment by Dr-Bean** [comment]
One more question: is PleyPy able to update itself?

**Comment by Dr-Bean** [comment]
And add a `BETA = 1` in the Makefile as well, forgot to mention that.

**Comment by mellowpimp** [comment]
@Dr-Bean Yes, if git is installed, it can fetch updates from the creators repo. 

2016-04-20 14:37:27	DEBUG	Requesting URL via GET method: https://api.github.com/repos/drzoidberg33/plexpy/commits/master?access_token=********

File: spk/plexpy-custom/src/config.ini (hunk 6 above)
Comment (rjblake) [inline spk/plexpy-custom/src/config.ini]: Have changed to 8010 if I understand correctly, although I see that most of the other packages (1 or odd numbers for http), don't follow the convention (perhaps came after). PlexPy default port is 8181

File: spk/plexpy-custom/Makefile (hunk 5 above)
Comment (rjblake) [inline spk/plexpy-custom/Makefile]: changed

**Comment by rjblake** [comment]
Have to check filesystem - other changes made

File: spk/plexpy-custom/src/config.ini (hunk 6 above)
Comment (Dr-Bean) [inline spk/plexpy-custom/src/config.ini]: Yep, older packages don't necessarily conform to that convention. Not much we can do about it without introducing issues, but we do want to enforce it for new packages.

**Comment by rjblake** [comment]
Have checked filesystem and made changes. No sc-media group needed. However, need to amend installer.sh to create a default config.ini (otherwise port defaults to 8181).

**Comment by rjblake** [comment]
@Dr-Bean - what's the next step? We waiting on a DSM6.0 adduser solution before any merge of new packages?

File: spk/plexpy-custom/src/installer.sh [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+    fi
+
+    # Create user
+    adduser -h ${INSTALL_DIR}/var -g "${DNAME} User" -G ${GROUP} -s /bin/sh -S -D ${USER}
+    
+    # get IP
+    sIPNAS=`/usr/syno/sbin/synonet --show | grep -m 1 IP:  | awk -F: '{gsub(/[ \t]+/, "", $2); print $2}'`
+    
+    # Add Host IP to config.ini
+    echo "http_host = ${sIPNAS}" >> ${INSTALL_DIR}/var/{CFG_FILE}
```
Comment (Dr-Bean) [inline spk/plexpy-custom/src/installer.sh]: The install documention doesn't show this is necessary afaict? If this doesn't need to be set, let's not.

If a value must be set, then does `0.0.0.0` work so it listens on all interfaces? 
Also, if a value must be set, then this should be moved to be part of the `if [ "${SYNOPKG_PKG_STATUS}" == "INSTALL" ]; then` on L42, so it's only done once on install.

File: spk/plexpy-custom/src/installer.sh [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+}
+
+preuninst ()
+{
+    # Stop the package
+    ${SSS} stop > /dev/null
+
+    # Remove the user if uninstalling
+    if [ "${SYNOPKG_PKG_STATUS}" == "UNINSTALL" ]; then
+
```
Comment (Dr-Bean) [inline spk/plexpy-custom/src/installer.sh]: Remove the empty line here

**Comment by Dr-Bean** [comment]
@rjblake Wrt your last question, no. We have enough packages that don't work on DSM6, one more doesn't matter that much. I'm not necessarily blocking packages/merges because of it, it's a lack of time on my end that's causing delays.
I added two more comments, please take a look. The `delgroup` you removed (the empty line I referred to) can be left in, it doesn't do any harm and probably is in sync with the rest of the packages using the `nobody` group.

**Comment by Dr-Bean** [comment]
And please squash commits after the last comments are handled, I think it's good to go with that :)

File: spk/plexpy-custom/src/installer.sh (hunk 8 above)
Comment (rjblake) [inline spk/plexpy-custom/src/installer.sh]: @Dr-Bean - I added this as there was an issue when PlexPy ran without an IP and a config.ini file that already included an assigned port. I'll double check why I did this, or if no longer required, remove the detection and assigning of the IP address. I do need to keep a config.ini with the `web_port = 8010`, or the default installation will assign a default port (8183 from memory).

**Comment by rjblake** [comment]
@Dr-Bean - will have to figure out how to 'squash commits' as don't seem to have the option available here on github.com (guess I need to do on my Docker (development) environment).

File: spk/plexpy-custom/src/installer.sh (hunk 8 above)
Comment (Dr-Bean) [inline spk/plexpy-custom/src/installer.sh]: The problem with setting a fixed IP in the config is that the package stops working correctly (from a user's perpective) when the IP of their NAS changes, or when multiple interfaces are active.
The sourcecode seems to confirm that: https://github.com/drzoidberg33/plexpy/blob/master/plexpy/config.py#L192.

Comment (rjblake) [inline spk/plexpy-custom/src/installer.sh]: I'll remove and retest. Didn't think about IP address changes as always use fixed IPs with my NAS for simplicity (as well as other applications requiring a fixed IP), but see the issue.

**Comment by rjblake** [comment]
@Dr-Bean - have tried to squash on the Docker container (from `docker pull synocommunity/spksrc`), but it keeps coming back with errors saying it cannot open the editor. As it turns out, there does not appear to be any editor (vi, vim, or other I can find) in the container. Any thoughts?

**Comment by Dr-Bean** [comment]
Not squashed, no. I can recreate the PR in one commit, or maybe I'll merge via a separate branch, I'll see when I get to it. I'll slap a label on it, so we know it's good to go.
Thanks for taking the time :)

**Comment by Dr-Bean** [comment]
Back to `Needs review`.
The web port is fixed at 8181 (`web_port` vs `http_port`, I'd think), and either I'm missing something, or the installer still sets a fixed IP in the config. Did the a later commit get lost in translation maybe?

**Comment by rjblake** [comment]
I'll check this out with a fresh install. The package I created previously installed on my Syno with port at 8010; so something has gone amiss

**Comment by rjblake** [comment]
@Dr-Bean - it looks like something changed in the config and it should now be `http_port` and not `web_port`as noted

**Comment by sgovaerts** [comment]
@rjblake @cytec Since Dr-Bean hasn't been active for a while will this remain inactive or is this something you can do or?

**Comment by KaraokeStu** [comment]
I would suggest that we try to squash all of these commits into a single commit and then check everything works as expected, once it all builds and we have confirmation you can both install and upgrade with the correct configuration staying in tact, we can look at adding back the "ready to merge" flag and getting it in there.

**Comment by cytec** [comment]
whoops... seems like i've totally missed that one... i'll have a look tomorrow but for a quick scrim it seems pretty good to me except 1 or 2 little changes i'd like to make... and squashing those commits of course ;)

**Comment by cytec** [comment]
@rjblake are you still up for this? or should i change stuff myself and pull from my branch?
if you are interested: i have 2 things left:

1. remove config file creation from installer.sh and move it to a seperate file in src see sickbeard-custom: https://github.com/SynoCommunity/spksrc/blob/master/spk/sickbeard-custom/src/config.ini
2. squash commits

**Comment by rjblake** [comment]
@cytec - I'm busy travelling so can't get to this till the new year. Feel
free to update/amend as you see fit

**Comment by cytec** [comment]
@rjblake well most likely i don't have time for this till January either... but if i have some and you haven't i'll fix it and merge. Thanks and have a good time ;)

**Comment by cytec** [comment]
I've created a new PR so i'll close this one thanks @rjblake

**Comment by rjblake** [comment]
@cytec thanks for that - haven't been near a computer in a couple of weeks

**Comment by rjblake** [comment]
Not sure what you did, but you killed my install! Why did you use the same
name and change developer to yourself? Now I have to work through why its
not loading

**Comment by rjblake** [comment]
No worries - I've reverted back to my own SPK and its working

**Comment by cytec** [comment]
I've just took your PR did a rebase and changed the 2 things i mentioned in this PR.

**Pull request closed without merging**